src/Bicep.Core/Registry/OciModuleRegistry.cs

```diff
@@ -95,6 +95,11 @@ public override bool IsArtifactRestoreRequired(OciModuleReference reference)
                 !this.FileResolver.FileExists(this.GetModuleFileUri(reference, ModuleFileType.Metadata));
         }
 
+        private InvalidModuleException GetUnknownArtifactException(string gotArtifactMediaType)
+                => new(
+                    $"Unknown ArtifactType: '{gotArtifactMediaType}'. Supported OCI artifactType fields are: (1) null or '{BicepMediaTypes.BicepModuleArtifactType}' for modules, or (2) '{BicepMediaTypes.BicepProviderArtifactType}' for resource type providers. {NewerVersionMightBeRequired}",
```

Comment:
I'd rather not mention null, it's only there for backwards compat and should never be used anymore.